synapse/handlers/sync.py

```diff
@@ -630,10 +633,17 @@ async def _load_filtered_recents(
 
             prev_batch_token = now_token.copy_and_replace("room_key", room_key)
 
+        # Don't bother to bundle aggregations if the timeline is unlimited,
+        # as clients will have all the necessary information.
+        bundled_aggregations = None
+        if limited:
```

Comment:
On +645 we effectively replace `limited` with `limited or newly_joined_room`. I'm not sure why; but might it make sense to do the same here?

Reply:
I don't think it does, but I'll double check.

Reply:
The previous code used what gets set as `limited` on `TimelineBatch`, so I think you're right here! 👍